### PR TITLE
don't allow loading packages from LOAD_PATH when testing or building

### PR DESCRIPTION
**Pull request by KristofferC**
also put deps in Manifest even for stdlibs and use them in the resolver

### CODE REVIEW

**Comment by StefanKarpinski** [comment]
We should do this, but I don't believe it's ever passed tests...

**Comment by KristofferC** [comment]
Sure, it did. It had a straight flush at https://github.com/JuliaLang/julia/pull/27572 when I said it was ready for review.

**Comment by StefanKarpinski** [comment]
I don't recall ever seeing it in a passing, mergeable state. When it gets there again, please merge.

File: src/Operations.jl [hunk 1]
```diff
@@ -146,7 +146,7 @@ function collect_project!(ctx::Context, pkg::PackageSpec, path::String, fix_deps
     if haskey(project, "version")
         pkg.version = VersionNumber(project["version"])
     else
-        @warn "project file for $(pkg.name) is missing a `version` entry"
+        # @warn "project file for $(pkg.name) is missing a `version` entry"
```
Comment (StefanKarpinski) [inline src/Operations.jl]: Post-review: how come this is commented out now?

Comment (KristofferC) [reply src/Operations.jl]: Sorry, it was spammy and made people too panicky about Project files.

**Pull request merged**